assets/js/util/index.js

```diff
@@ -207,17 +207,21 @@ export const numberFormat = ( number, options = {} ) => {
 /**
  * Gets the current locale for use with browser APIs.
  *
+ * @param {Object} _global The global window object.
+ *
  * @return {string} Current Site Kit locale if set, otherwise the current language set by the browser.
  *                  E.g. `en-US` or `de-DE`
  */
-export const getLocale = () => {
-	const siteKitLocale = get( global, [ 'googlesitekit', 'locale', '', 'lang' ] );
-
+export const getLocale = ( _global = global ) => {
+	const siteKitLocale = get( _global, [ 'googlesitekit', 'locale', '', 'lang' ] );
 	if ( siteKitLocale ) {
-		return siteKitLocale.replace( '_', '-' );
+		const matches = siteKitLocale.match( /(\w{2})?(_)?(\w{2})/ );
```

Comment:
This should maybe be more strict to match only at the beginning of the string. Probably not a big deal for the values that WordPress will pass to it, but overall that would be more error-proof.

Also right now this will match a string like "dede" or "deDE" which it probably shouldn't. Both of these aren't blockers, but could be improved.